### PR TITLE
Add docment for TfmdDL

### PR DESCRIPTION
**Pull request by malligaraj**
Add `docment` for the following:
- class TfmdDL
- show_batch()
- show_results()

This PR implements `TfmdDL` in #8

### CODE REVIEW

File: fastai/data/core.py [hunk 1]
```diff
@@ -59,7 +82,11 @@ def _retain_dl(self,b):
         return retain_types(b, typs=self._types)
 
     @delegates(DataLoader.new)
-    def new(self, dataset=None, cls=None, **kwargs):
+    def new(self,
+        dataset=None, # `Datasets` object
```
Comment (kevinbird15) [inline fastai/data/core.py:86]:
```suggestion
        dataset=None, # Map- or iterable-style dataset from which to load the data
```

File: fastai/data/core.py [hunk 2]
```diff
@@ -38,7 +53,15 @@ def show_results(x, y, samples, outs, ctxs=None, max_n=9, **kwargs):
 @delegates()
 class TfmdDL(DataLoader):
     "Transformed `DataLoader`"
-    def __init__(self, dataset, bs=64, shuffle=False, num_workers=None, verbose=False, do_setup=True, **kwargs):
+    def __init__(self,
+        dataset, # `Datasets` object
```
Comment (kevinbird15) [inline fastai/data/core.py:57]:
```suggestion
        dataset, # Map- or iterable-style dataset from which to load the data
```

File: fastai/data/core.py [hunk 3]
```diff
@@ -38,7 +53,15 @@ def show_results(x, y, samples, outs, ctxs=None, max_n=9, **kwargs):
...
+        bs:int=64, # Size of batch
+        shuffle:bool=False, # Whether to shuffle data
+        num_workers:int=None, # Number of CPU cores to use in parallel
```
Comment (kevinbird15) [inline fastai/data/core.py:60]:
```suggestion
        num_workers:int=None, # Number of CPU cores to use in parallel (default: All available up to 16) 
```

**Review by kevinbird15 — CHANGES_REQUESTED** [review]
A few small changes.  Wasn't entirely sure what to put for `dataset` so feel free to play around with that

**Review by kevinbird15 — APPROVED** [review]
Looks good to me

File: fastai/data/core.py [hunk 4]
```diff
@@ -20,7 +27,15 @@ def show_batch(x, y, samples, ctxs=None, max_n=9, **kwargs):
 
 # Cell
 @typedispatch
-def show_results(x, y, samples, outs, ctxs=None, max_n=9, **kwargs):
+def show_results(
+    x, # Input(s) in the batch
+    y, # Target(s) in the batch
+    samples, # List of (`x`, `y`) pairs of length `max_n`
+    outs, # List of predicted output(s) from the model
+    ctxs=None, # List of `ctx` objects to show data. Could be matplotlib axis, DataFrame etc
```
Comment (tmabraham) [inline fastai/data/core.py:35]:
```suggestion
    ctxs=None, # List of `ctx` objects to show data. Could be a matplotlib axis, DataFrame, etc.
```

File: fastai/data/core.py [hunk 5]
```diff
@@ -9,7 +9,14 @@
 
 # Cell
 @typedispatch
-def show_batch(x, y, samples, ctxs=None, max_n=9, **kwargs):
+def show_batch(
+    x, # Input(s) in the batch
+    y, # Target(s) in the batch
+    samples, # List of (`x`, `y`) pairs of length `max_n`
+    ctxs=None, # List of `ctx` objects to show data. Could be matplotlib axis, DataFrame etc
+    max_n=9, # Maximum number of `samples` to show
```
Comment (tmabraham) [inline fastai/data/core.py:17]:
```suggestion
    max_n:int=9, # Maximum number of `samples` to show
```

File: fastai/data/core.py [hunk 6]
```diff
@@ -20,7 +27,15 @@ def show_batch(x, y, samples, ctxs=None, max_n=9, **kwargs):
...
 # Cell
 @typedispatch
-def show_results(x, y, samples, outs, ctxs=None, max_n=9, **kwargs):
+def show_results(
+    x, # Input(s) in the batch
+    y, # Target(s) in the batch
+    samples, # List of (`x`, `y`) pairs of length `max_n`
+    outs, # List of predicted output(s) from the model
+    ctxs=None, # List of `ctx` objects to show data. Could be matplotlib axis, DataFrame etc
+    max_n=9, # Maximum number of `samples` to show
```
Comment (tmabraham) [inline fastai/data/core.py:36]:
```suggestion
    max_n:int=9, # Maximum number of `samples` to show
```

File: fastai/data/core.py [hunk 7]
```diff
@@ -9,7 +9,14 @@
 
 # Cell
 @typedispatch
-def show_batch(x, y, samples, ctxs=None, max_n=9, **kwargs):
+def show_batch(
```
Comment (tmabraham) [inline fastai/data/core.py:12]: Needs a docstring

File: fastai/data/core.py [hunk 8]
```diff
@@ -20,7 +27,15 @@ def show_batch(x, y, samples, ctxs=None, max_n=9, **kwargs):
 
 # Cell
 @typedispatch
-def show_results(x, y, samples, outs, ctxs=None, max_n=9, **kwargs):
+def show_results(
```
Comment (tmabraham) [inline fastai/data/core.py:30]: Needs a docstring

File: fastai/data/core.py [hunk 9]
```diff
@@ -93,7 +128,14 @@ def _pre_show_batch(self, b, max_n=9):
         if not is_listy(b): b,its = [b],L((o,) for o in its)
         return detuplify(b[:self.n_inp]),detuplify(b[self.n_inp:]),its
 
-    def show_batch(self, b=None, max_n=9, ctxs=None, show=True, unique=False, **kwargs):
+    def show_batch(self,
```
Comment (tmabraham) [inline fastai/data/core.py:131]: Needs docstring

File: fastai/data/core.py [hunk 10]
```diff
@@ -102,7 +144,14 @@ def show_batch(self, b=None, max_n=9, ctxs=None, show=True, unique=False, **kwar
         show_batch(*self._pre_show_batch(b, max_n=max_n), ctxs=ctxs, max_n=max_n, **kwargs)
         if unique: self.get_idxs = old_get_idxs
 
-    def show_results(self, b, out, max_n=9, ctxs=None, show=True, **kwargs):
+    def show_results(self,
```
Comment (tmabraham) [inline fastai/data/core.py:147]: needs docstring

File: fastai/data/core.py [hunk 11]
```diff
@@ -111,12 +160,14 @@ def show_results(self, b, out, max_n=9, ctxs=None, show=True, **kwargs):
         show_results(*res, ctxs=ctxs, max_n=max_n, **kwargs)
 
     @property
-    def n_inp(self):
+    def n_inp(self) -> int:
```
Comment (tmabraham) [inline fastai/data/core.py:163]: needs docstring

**Review by tmabraham — CHANGES_REQUESTED** [review]

File: fastai/data/core.py (hunk 7 above)
Comment (malligaraj) [reply fastai/data/core.py:12]: Done

File: fastai/data/core.py (hunk 5 above)
Comment (malligaraj) [reply fastai/data/core.py:17]: Adding type hint for max_n, failed a few check jobs. Failing check jobs in [5601bae](https://github.com/muellerzr/fastai-docment-sprint/pull/93/commits/5601baefc56a4e9e63e5ac7a5e92b2898503ee8d)

File: fastai/data/core.py (hunk 8 above)
Comment (malligaraj) [reply fastai/data/core.py:30]: Done

File: fastai/data/core.py (hunk 6 above)
Comment (malligaraj) [reply fastai/data/core.py:36]: Adding type hint for max_n, failed a few check jobs. Failing check jobs in [5601bae](https://github.com/muellerzr/fastai-docment-sprint/pull/93/commits/5601baefc56a4e9e63e5ac7a5e92b2898503ee8d)

File: fastai/data/core.py (hunk 9 above)
Comment (malligaraj) [reply fastai/data/core.py:131]: Done

File: fastai/data/core.py (hunk 10 above)
Comment (malligaraj) [reply fastai/data/core.py:147]: Done

File: fastai/data/core.py (hunk 11 above)
Comment (malligaraj) [reply fastai/data/core.py:163]: Done

**Review by tmabraham — APPROVED** [review]

**Pull request merged**